### DISCUSSION TITLE
Command option for each file

### DISCUSSION THREAD

**Discussion by jcook-uptycs** [evilmartians/lefthook#423] [Ideas]
I have found myself writing a bunch of for each loops in the `run` option. 

There are a few lint and testing tools that only take a single file, or directory as input. These then require I write a for each to take the {files} or {staged_files} and run the tool on each file. 

While this works, it makes for a very long config file. Also it affects the parallel option that lefthook has. Each of these for each is then no longer parallel with each other.

**Comment by mrexox** [comment]
Hey! What tools do you use? :thinking: Looks like a good option but it's quite hard to implement.

**Comment by jcook-uptycs** [comment]
I find it with a lot of tools. But here are a few. 

```yaml
pre-commit:
  commands:
    trivy-dockerfile:
      glob: "**/*{Dockerfile}*"
      env:
        TRIVY_EXIT_CODE: 1
      run: |
        for FILE in {files}
        do
          trivy filesystem \
            --license-full \
            --security-checks=vuln,secret,license \
            "${FILE}"
        done
    depcheck:
      glob: "*{package.json,package-lock.json,.js}"
      env:
        report: ./tmp/reports/depcheck.txt
      run: |
        set -eo pipefail
        mkdir -p "$(dirname "${report}")"
        echo > "${report}"
        for FILE in {files}
        do
          echo "scanning file '${FILE}'..." | tee -a "${report}"
          npx --yes depcheck \
            "$(dirname "${FILE}")" | tee -a "${report}" ||  EXIT=1
        done
        exit ${EXIT}
```

A lot of this might be because we have a bit of a mono repo setup.
